web/src/hooks/useRobotAccounts.ts

```diff
@@ -182,3 +188,248 @@ interface createNewRobotForNamespaceParams {
   selectedTeams: IRobotTeam[];
   robotDefaultPerm: string;
 }
+
+export function useRobotPermissions({orgName, robName, onSuccess, onError}) {
+  const [namespace, setNamespace] = useState(orgName);
+  const [robotName, setRobotName] = useState(robName);
+  const [page, setPage] = useState(1);
+  const [perPage, setPerPage] = useState(10);
+
+  const {usernames} = useOrganizations();
+  const isUser = usernames.includes(orgName);
+
+  const {
+    data: robotPermissions,
+    isLoading: loading,
+    error,
+  } = useQuery(
+    ['Namespace', namespace, 'robot', robotName, 'permissions'],
+    ({signal}) =>
+      fetchRobotPermissionsForNamespace(namespace, robotName, isUser, signal),
+    {
+      enabled: true,
+      placeholderData: [],
+      onSuccess: (result) => {
+        onSuccess(result);
+      },
+      onError: (err) => {
+        onError(err);
+      },
+    },
+  );
+
+  const queryClient = useQueryClient();
+  const deleteRepoPermsMutator = useMutation(
+    async (repoNames: string[]) => {
+      await bulkDeleteRepoPermsForRobot(namespace, robName, repoNames, isUser);
+    },
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries(['Namespace', namespace, 'robots']);
+        queryClient.invalidateQueries([
+          'Namespace',
+          namespace,
+          'robot',
+          robotName,
+          'permissions',
+        ]);
+        onSuccess();
+      },
+      onError: (err) => {
+        onError(err);
+      },
+    },
+  );
+
+  const updateRepoPermsMutator = useMutation(
+    async (repoPerms: IRepoPerm[]) => {
+      return await bulkUpdateRepoPermsForRobot(namespace, robotName, repoPerms);
+    },
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries(['Namespace', namespace, 'robots']);
+        queryClient.invalidateQueries([
+          'Namespace',
+          namespace,
+          'robot',
+          robotName,
+          'permissions',
+        ]);
+        onSuccess();
+      },
+      onError: (err) => {
+        onError(err);
+      },
+    },
+  );
+
+  return {
+    result: robotPermissions,
```

Comment:
It seems as though `result` is not being used? I see it being referenced [here](https://github.com/quay/quay/blob/master/web/src/components/modals/RobotRepositoryPermissions.tsx#L29), but it doesn't seem to have any effect